hs_labels/models.py

```diff
@@ -0,0 +1,876 @@
+    rlabels = models.ForeignKey('ResourceLabels', null=True, editable=False,
+                                related_name="rl2url",     # unused but must be defined and unique
+                                help_text='resource to which a label applies')
+    label = models.TextField()
+
+
+class UserLabels(models.Model):
+    """
+    Projection class puts methods and content inside basic User object
+    so that one can access things easily from that context.
+    This model is injected into the BaseResource as the related name "ulabels".
+    Thus for an User u, u.ulabels is this model.
+    """
+
+    user = models.OneToOneField(User,
+                                editable=False,
+                                null=True,
+                                related_name='ulabels',  # induced field in User class.
+                                related_query_name='ulabels')
+
+    ##########################################
+    ##########################################
+    # PUBLIC FUNCTIONS: resources
+    ##########################################
+    ##########################################
+
+    ##########################################
+    # get various aspects of labeling
+    ##########################################
+
+    @property
+    def labeled_resources(self):
+        """
+        Get a list of resources labeled by a user.
+
+        This list eliminates duplicates.
+
+        :return: List of resource objects labeled by user.
+        """
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self,
+                                           rlabels__rl2url__kind=LabelCodes.LABEL).distinct()
+
+    @property
+    def filed_resources(self):
+        """
+        Get a list of resources filed by a user.
+
+        This list eliminates duplicates.
+
+        :return: List of resource objects assigned to folders by a user.
+        """
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self,
+                                           rlabels__rl2url__kind=LabelCodes.FOLDER).distinct()
+
+    @property
+    def favorited_resources(self):
+        """
+        Get a list of resources favorited by a user.
+
+        This list eliminates duplicates.
+
+        :return: List of resource objects marked as favorite by a user.
+        """
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self,
+                                           rlabels__rl2url__kind=LabelCodes.FAVORITE).distinct()
+
+    @property
+    def my_resources(self):
+        """
+        Get a list of resources marked as mine (add to my resources) by a user.
+
+        This list eliminates duplicates.
+
+        :return: List of resource objects marked as mine by a user.
+        """
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self,
+                                           rlabels__rl2url__kind=LabelCodes.MINE).distinct()
```

Comment:
That may be your convention for properties. If so, I was unaware of it. I'm not sure that is the proper interpretation, either. I am sure that there are other properties that are simply computed from the class instance.